src/compiler/factory.ts

```diff
@@ -827,6 +824,7 @@ namespace ts {
         // Create an identifier and give it a parent. This allows us to resolve the react
         // namespace during emit.
         const react = createIdentifier(reactNamespace || "React");
+        react.flags &= ~NodeFlags.Synthesized;

```

Comment:
No, this is a specific case to ensure the react namespace is properly resolved by the emit resolver, which would otherwise try to find an original source tree node that doesn't exist.